sklearn/preprocessing/data.py

```diff
@@ -3016,24 +3019,18 @@ def _fit_transform(self, X):
         indices = np.cumsum(n_values)
         self.feature_indices_ = indices
 
-        column_start = np.tile(indices[:-1], n_samples)
-        column_end = (X + indices[:-1]).ravel()
-        column_indices = np.hstack([np.arange(s, e) for s, e
-                                   in zip(column_start, column_end)])
-        row_indices = np.repeat(np.arange(n_samples, dtype=np.int32),
-                                X.sum(axis=1))
-        data = np.ones(X.sum())
-        out = sparse.coo_matrix((data, (row_indices, column_indices)),
-                                shape=(n_samples, indices[-1]),
-                                dtype=self.dtype).tocsr()
+        mask = (X < self.n_values_).ravel()
+        if np.any(~mask):
+            if self.handle_greater == 'error':
+                raise ValueError("unknown ordinal feature present %s "
```

Comment:
how about "handle_greater='error' but feature value %d > n_values_[%d] = %d" and just give a single example...?

Reply:
By give single example do you mean, if there are multiple feature value which are failing, just raise the error printing any one of them?

Reply:
I think that's at least as useful as printing all of them. Alternatively, state the number of feature values that exceed `n_values_`

Reply:
I am going ahead with number of feature values that exceed n_values_. Just a thought, should i also include the row and col indices of those feature values? Or maybe thats too verbose?

Reply:
Too verbose, IMO. Sometimes it's helpful to give an example, but I think this is straightforward enough that the user can work it out